packages/webviz-core/src/panels/Teleop/index.js

```diff
@@ -0,0 +1,291 @@
+  static panelType = "Teleop";
+  static defaultConfig = {
+    buttonColor: "#00A871",
+    topicName: "/cmd_vel",
+  };
+
+  _publisher = React.createRef<Publisher>();
+
+  state = {
+    cachedProps: {},
+    datatypeNames: [],
+    error: null,
+    parsedObject: undefined,
+    pressing: {ArrowLeft: false, ArrowRight: false, ArrowUp: false, ArrowDown: false},
+  };
+
+  static getDerivedStateFromProps(props: Props, state: PanelState) {
+    const newState: $Shape<PanelState> = parseInput(props.config.value);
+    let changed = false;
+
+    if (props !== state.cachedProps) {
+      newState.cachedProps = props;
+      changed = true;
+    }
+
+    if (props.datatypes !== state.cachedProps.datatypes) {
+      newState.datatypeNames = Object.keys(props.datatypes).sort();
+      changed = true;
+    }
+
+    return changed ? newState : null;
+  }
+
+  composeTwist = (key) => {
+    const COMMANDS = { // Map x, y, z, th here
+      ArrowUp: [1, 0, 0, 0], 
+      ArrowDown: [-1, 0, 0, 0], 
+      ArrowLeft: [0, 0, 0, 1], 
+      ArrowRight: [0, 0, 0, -1],
+      " ": [0, 0, 0, 0], // Stop
+    };
+    const [x, y, z, th] = COMMANDS[key];
+    const speed = 0.5, turn = 1.0;
+    return {
+      linear: {
+        x: x*speed,
+        y: y*speed,
+        z: z*speed
+      },
+      angular: {
+        x: 0,
+        y: 0,
+        z: th * turn
+      }
+    }
+  };
+
+  _onChange = (event: SyntheticInputEvent<HTMLTextAreaElement>) => {
+    this.props.saveConfig({ value: event.target.value });
+  };
+
+  _onCommandButtonClick = (command) => {
+    // Simulate key press and release
+    this._handleKey({key: command, type: "keydown"});
+    this._handleKey({key: command, type: "keyup"});
+  };
+
+  _handleKey = (event) => {
+    this.setState(function(state, props) {
+      let newPressing = state.pressing;
+      newPressing[event.key] = event.type == "keydown";
+      if(this._publisher.current){
+        if(event.type=="keydown" ){
+          this._publisher.current.publish(this.composeTwist(event.key));
+        }
+      } else {
+        console.error("Publisher not set!");
+      }
+      return {
+        pressing: newPressing
+      };
+    });
+
+    this.forceUpdate(); // HACK Appears to be required to update the colors quickly???
+  };
+
+  // Add handler here for all desired keybindings (they are all the same!)
+  _keyHandlers = {
+    ArrowLeft: this._handleKey,
+    ArrowRight: this._handleKey,
+    ArrowUp: this._handleKey,
+    ArrowDown: this._handleKey,
+    " ": this._handleKey, // Space
+  };
+
+  _renderMenuContent() {
+    const { config, saveConfig } = this.props;
+
+    return (
+      <>
+        <Item>
+          <PanelToolbarLabel>Button color (rgba or hex)</PanelToolbarLabel>
+          <PanelToolbarInput
+            type="text"
+            value={config.buttonColor}
+            onChange={(event) => {
+              saveConfig({ buttonColor: event.target.value });
```

Comment:
Curious what's your use case for customizing button color?

Reply:
Nothing actually, it's a remnant from the Publish panel from which I pulled the original code. I deleted it.